### ISSUE TITLE
⚠︎ Local/Global ESLint binary not found. Fallback to packaged one. Set binary in preferences.

### ISSUE THREAD

**Issue by MoOx** [AtomLinter/linter-eslint#162]
I keep having this issue but I have correctly eslint installed. In each opened folders.

![screen shot 2015-07-13 at 08 10 41](https://cloud.githubusercontent.com/assets/157534/8643631/bc1fced2-2936-11e5-823c-3ef6e7322bfe.png)

![screen shot 2015-07-13 at 08 10 33](https://cloud.githubusercontent.com/assets/157534/8643630/bc1c2aac-2936-11e5-92a2-6eb5f2b011a0.png)

I got this warning each time I press a key on my keyboard, that's annoying.

I didn't get this issue in the past so no idea what happened...

FYI, here is the output of the Atom console

```
/Users/MoOx/setup/dotfiles/atom/packages/linter-jsonlint/lib/init.coffee:14 activate linter-jsonlint
index.js:56 Window load time: 19035ms
linter-eslint.js:126 [Linter-ESLint] no output from CLI
linter-eslint.js:127 eslint binary: /Users/MoOx/setup/dotfiles/atom/packages/linter-eslint/node_modules/.bin/eslint
linter-eslint.js:128 params: --stdin-filename /Users/MoOx/Sync/Development/postcss/selector-matches/src/index.js --config /Users/MoOx/Sync/Development/postcss/selector-matches/.eslintrc --format /Users/MoOx/setup/dotfiles/atom/packages/linter-eslint/node_modules/eslint-json/json.js --stdin
linter-eslint.js:129 cwd: /Users/MoOx/setup/dotfiles/atom/packages/linter-eslint/lib
linter-eslint.js:126 [Linter-ESLint] no output from CLI
linter-eslint.js:127 eslint binary: /Users/MoOx/setup/dotfiles/atom/packages/linter-eslint/node_modules/.bin/eslint
linter-eslint.js:128 params: --stdin-filename /Users/MoOx/Sync/Development/postcss/selector-matches/src/index.js --config /Users/MoOx/Sync/Development/postcss/selector-matches/.eslintrc --format /Users/MoOx/setup/dotfiles/atom/packages/linter-eslint/node_modules/eslint-json/json.js --stdin
linter-eslint.js:129 cwd: /Users/MoOx/setup/dotfiles/atom/packages/linter-eslint/lib
events.js:141 Uncaught Error: write EPIPE
linter-eslint.js:126 [Linter-ESLint] no output from CLI
linter-eslint.js:127 eslint binary: /Users/MoOx/setup/dotfiles/atom/packages/linter-eslint/node_modules/.bin/eslint
linter-eslint.js:128 params: --stdin-filename /Users/MoOx/Sync/Development/postcss/custom-selectors/src/index.js --config /Users/MoOx/Sync/Development/postcss/custom-selectors/.eslintrc --format /Users/MoOx/setup/dotfiles/atom/packages/linter-eslint/node_modules/eslint-json/json.js --stdin
linter-eslint.js:129 cwd: /Users/MoOx/setup/dotfiles/atom/packages/linter-eslint/lib
linter-eslint.js:126 [Linter-ESLint] no output from CLI
linter-eslint.js:127 eslint binary: /Users/MoOx/setup/dotfiles/atom/packages/linter-eslint/node_modules/.bin/eslint
linter-eslint.js:128 params: --stdin-filename /Users/MoOx/Sync/Development/postcss/custom-selectors/src/index.js --config /Users/MoOx/Sync/Development/postcss/custom-selectors/.eslintrc --format /Users/MoOx/setup/dotfiles/atom/packages/linter-eslint/node_modules/eslint-json/json.js --stdin
linter-eslint.js:129 cwd: /Users/MoOx/setup/dotfiles/atom/packages/linter-eslint/lib
linter-eslint.js:126 [Linter-ESLint] no output from CLI
linter-eslint.js:127 eslint binary: /Users/MoOx/setup/dotfiles/atom/packages/linter-eslint/node_modules/.bin/eslint
linter-eslint.js:128 params: --stdin-filename /Users/MoOx/Sync/Development/postcss/custom-selectors/src/index.js --config /Users/MoOx/Sync/Development/postcss/custom-selectors/.eslintrc --format /Users/MoOx/setup/dotfiles/atom/packages/linter-eslint/node_modules/eslint-json/json.js --stdin
linter-eslint.js:129 cwd: /Users/MoOx/setup/dotfiles/atom/packages/linter-eslint/lib
linter-eslint.js:126 [Linter-ESLint] no output from CLI
... (same stuff on loooots of line)
```

Any idea ?

**Comment by keithamus** [comment]
I have the same issue.

In part I think this may be because the PR in #145 seems to have disappeared. I can't figure out why exactly.

**Comment by keithamus** [comment]
@iam4x I think this project could really do with some unit tests, to prevent these kinds of errors. It seems #145 PR has just disappeared, and #145 PR is broken anyway - as it never resolves. Thoughts?

**Comment by globexdesigns** [comment]
Having the same issue after upgrading to 2.4.1

**Comment by andykenward** [comment]
Having same issue. Tried to downgrade but that failed :(

**Comment by astephenb** [comment]
I am having the same issue. It appears to be related to opening atom from the app launcher instead of CLI with a target. It seems that eslint is initiated without a cwd and gets defaulted to the package directory to search for the binary.

**Comment by targeter21** [comment]
I am seeing this issue and I start Atom from the command line. My Atom is up to date with the linter and  linter-eslint packages also fully up to date.

**Comment by LeoAJ** [comment]
+ 1

**Comment by LeoAJ** [comment]
+1

**Comment by florrain** [comment]
+1, annoying.

**Comment by florrain** [comment]
Even manually setting the binary makes it worse:

<img width="451" alt="screen shot 2015-07-13 at 20 00 13" src="https://cloud.githubusercontent.com/assets/2470945/8665564/1a346bdc-299a-11e5-9da3-20d73ca5aca3.png">

I installed eslint globally then `which eslint` gave me that path I used: `/Users/flo/.nvm/versions/node/v0.12.4/bin/eslint`

**Comment by unlok** [comment]
+1

**Comment by iam4x** [comment]
Related to #149, let's keep the discussion in one place.
